src/porepy/models/abstract_model.py

```diff
@@ -345,49 +346,62 @@ def _export(self) -> None:
         """
         pass
 
-    def _domain_boundary_sides(
-        self,
-        sd: pp.Grid,
-        tol: float = 1e-10,
-    ) -> tuple[
-        np.ndarray,
-        np.ndarray,
-        np.ndarray,
-        np.ndarray,
-        np.ndarray,
-        np.ndarray,
-        np.ndarray,
-    ]:
+    def _domain_boundary_sides(self, sd: pp.Grid, tol: float = 1e-10) -> NamedTuple:
         """Obtain indices of the faces lying on the sides of the domain boundaries.
 
         The method is primarily intended for box-shaped domains. However, it can also be
         applied to non-box-shaped domains (e.g., domains with perturbed boundary nodes)
         provided `tol` is tuned accordingly.
 
-        Args:
+        Parameters:
             sd: Subdomain grid.
             tol: Tolerance used to determine whether a face center lies on a boundary side.
 
         Returns:
-            Tuple of boolean arrays, each one of size sd.num_faces. For a given array,
-                a True element indicates that the face lies on the side of the domain.
-                Otherwise, the element is set to False.
+            NamedTuple containing the domain boundary sides. Available attributes are:
+
+                - all_bf (np.ndarray of int): indices of the boundary faces.
+                - east (np.ndarray of bool): flags of the faces lying on the East side.
+                - west (np.ndarray of bool): flags of the faces lying on the West side.
+                - north (np.ndarray of bool): flags of the faces lying on the North side.
+                - south (np.ndarray of bool): flags of the faces lying on the South side.
+                - top (np.ndarray of bool): flags of the faces lying on the Top side.
+                - bottom (np.ndarray of bool): flags of the faces lying on Bottom side.
+
+        Examples:
+
+            .. code:: python
+
+                model = pp.IncompressibleFlow({})
+                model.prepare_simulation()
+                sd = model.mdg.subdomains()[0]
+                sides = model._domain_boundary_sides(sd)
+                print(sides.north)
```

Comment:
```suggestion
# Access north faces using index or name is equivalent:
north_by_index = sides[3]
north_by_name = sides.north
assert north_by_index == north_by_name
```

Reply:
Great improvement. I've incorporated it in a separate commit.